pkg/tracing/provider/factory.go

```diff
@@ -0,0 +1,59 @@
+package provider
+
+import (
+	"context"
+	"io"
+	"io/ioutil"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/improbable-eng/thanos/pkg/tracing/provider/jaeger"
+	"github.com/improbable-eng/thanos/pkg/tracing/provider/stackdriver"
+	"github.com/opentracing/opentracing-go"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+type TracingProvider string
+
+const (
+	STACKDRIVER TracingProvider = "STACKDRIVER"
+	JAEGER      TracingProvider = "JAEGER"
+)
+
+type TracingConfig struct {
+	Type   TracingProvider `yaml:"type"`
+	Config interface{}     `yaml:"config,omitempty"`
+}
+
+func NewTracer(ctx context.Context, logger log.Logger, confContentYaml []byte) (opentracing.Tracer, io.Closer, error) {
+	level.Info(logger).Log("msg", "loading tracing configuration")
+	tracingConf := &TracingConfig{}
+
+	if err := yaml.UnmarshalStrict(confContentYaml, tracingConf); err != nil {
+		return nil, nil, errors.Wrap(err, "parsing config tracing YAML")
+	}
+
+	var config []byte
+	var err error
+	if tracingConf.Config != nil {
```

Comment:
Can it be nil?

Reply:
if config in YAML empty on unset. And marshal(nil) != nil